src/graph/pickle.cc

```diff
@@ -97,6 +99,8 @@ HeteroGraphPtr HeteroUnpickle(const HeteroPickleStates& states) {
   std::vector<HeteroGraphPtr> relgraphs(metagraph->NumEdges());
   std::vector<int64_t> num_nodes_per_type;
   CHECK(strm->Read(&num_nodes_per_type)) << "Invalid num_nodes_per_type";
+  bool is_pinned = false;
+  CHECK(strm->Read(&is_pinned)) << "Invalid flag 'is_pinned'";
```

Comment:
I think directly modifying and `HeteroUnpickle` will break unpickling from existing binaries.  You will need to version it by
* Skip reading the `is_pinned` flag (and set it to False) when pickle state version is 1.
* Set the pickle state version to 2 in `_CAPI_DGLCreateHeteroPickleStates`.
You could test compatibility by creating a pickle file in 0.8.0post2, and try unpickling it with your new implementation.

Reply:
I tried to set state version in `_CAPI_DGLCreateHeteroPickleStates`, but seems `_CAPI_DGLCreateHeteroPickleStates` is called in `pickle.load()`. Is this expected?

Reply:
Oh you are right.  `pickle.load()` will first create a `HeteroPickleStates` and then populate it via `__setstate__`.  `version` should be a parameter that is loaded from `HeteroPickleStates.__setstate__` method where the `state` argument is unpacked (the first element of `state` is version number).

I also noticed that the version number is never actually stored correctly in the pickle file since `HeteroPickle` C function never set the number.  Could you check if it's indeed the case?  If so then we will also need to fix that.

Reply:
Confirmed. `HeteroPickleStates.version` is `0`, not the expected `1`. One question: if `g.pkl` whose version is `1` is loaded in new DGL which has version `2`, when dump with new DGL, the dumped version is expected to be `2` or `1`(the old one).

Reply:
The output should always be the new one (2).